src/app/components/ATIAnalytics/params/genericPage/buildParams.test.ts

```diff
@@ -0,0 +1,71 @@
+import * as analyticsUtils from '#lib/analyticsUtils';
+import { RequestContextProps } from '../../../../contexts/RequestContext';
+import { ServiceConfig } from '../../../../models/types/serviceConfig';
+import { buildPageATIParams, buildPageATIUrl } from './buildParams';
+
+(analyticsUtils.getAtUserId as jest.Mock) = jest.fn();
+(analyticsUtils.getCurrentTime as jest.Mock) = jest
+  .fn()
+  .mockReturnValue('00-00-00');
+(analyticsUtils.getPublishedDatetime as jest.Mock) = jest
+  .fn()
+  .mockReturnValue('1970-01-01T00:00:00.000Z');
+
+// @ts-expect-error - only partial data required for testing purposes
+const requestContext: RequestContextProps = {
+  platform: 'canonical',
+  statsDestination: 'statsDestination',
+  id: 'validId',
+};
+
+// @ts-expect-error - only partial data required for testing purposes
+const serviceContext: ServiceConfig = {
+  atiAnalyticsAppName: 'atiAnalyticsAppName',
+  atiAnalyticsProducerId: 'atiAnalyticsProducerId',
+  service: 'pidgin',
+  lang: 'pcm',
+};
+
+const atiData = {
+  analytics: {
+    contentId: 'urn:bbc:tipo:topic:cm7682qz7v1t',
+    contentType: 'index-home',
+    pageIdentifier: 'kyrgyz.page',
+  },
+  title: 'pageTitle',
+};
+
+const validPageURLParams = {
+  appName: serviceContext.atiAnalyticsAppName,
+  contentId: atiData.analytics.contentId,
+  producerId: serviceContext.atiAnalyticsProducerId,
```

Comment:
```suggestion
const validPageURLParams = {
  appName: 'atiAnalyticsAppName',
  contentId: 'index-home',
  producerId: 'atiAnalyticsProducerId',
  //.. string literals for remaining keys
```
Can't cite a reference but a previous colleague put me onto a good habit for tests, don't do DRY, _Do_ Repeat Yourself. Here you are using the objects going into the test to assert the expected output, it's isn't super clear what the expected output is 'at a glance' and any mistakes made in your config objects will not be picked up in your assertions. Not biggee but a personal approach I'm a fan of